### PR TITLE
 make eltType mandatory first arg in RandomStream ctors

### PR DESCRIPTION
**Pull request by mppf**
Fixed issue #5281 by adjusting RandomStream constructor arguments. Before this PR, they are
``` chapel
      proc RandomStream(seed: int(64) = SeedGenerator.currentTime,
                        param parSafe: bool = true,
                        type eltType = real(64) )
```
and this PR changes them to
``` chapel
      proc RandomStream(type eltType,
                        seed: int(64) = SeedGenerator.currentTime,
                        param parSafe: bool = true)
```

since:

1) the type is more important, important enough to
    be the first argument and important enough to be required
2) the default of `real` only really made sense when NPB
    was the only RNG and it couldn't do integers.

(See the issue for details for why this change resolves the issue).

This PR additionally uses a function of last resort (with the "compiler generated" pragma) to create a deprecation error in cases that would require code changes.

Passed full local testing.
Passed quickstart+GASNet testing.

Reviewed by @ben-albrecht - thanks!

### CODE REVIEW

File: modules/standard/Random.chpl [hunk 1]
```diff
@@ -2029,19 +2057,54 @@ module Random {
...
         :type parSafe: `bool`
 
-        :arg eltType: The element type to be generated.  Defaults to `real(64)`.
-        :type eltType: `type`
       */
+      proc NPBRandomStream(type eltType,
+                           seed: int(64) = SeedGenerator.oddCurrentTime,
+                           param parSafe: bool = true) {
+
+        // The mod operation is written in these steps in order
```
Comment (ben-albrecht) [inline modules/standard/Random.chpl]: Is this a fix unrelated to the constructor argument re-ordering?

File: modules/standard/Random.chpl [hunk 2]
```diff
@@ -210,6 +210,17 @@ module Random {
...
   }
 
+  // remove this deprecation version when appropriate
+  pragma "no doc"
+  pragma "compiler generated"
+  proc makeRandomStream(seed: int(64) = SeedGenerator.oddCurrentTime,
+                        param parSafe: bool = true,
+                        type eltType = real(64),
+                        param algorithm = defaultRNG) {
+    compilerWarning("makeRandomStream now requires eltType as first argument");
```
Comment (ben-albrecht) [inline modules/standard/Random.chpl]: Really minor, but removing the word 'now' would make all of the warnings consistent.

**Comment by ben-albrecht** [comment]
Aside from a really minor wording suggestion, it looks good. Thanks for the fix.

**Comment by bradcray** [comment]
Sorry that I didn't get a chance to look at this sooner.  I think that the warning that's printed when the old constructors/routines are used should use the term "deprecated" to give a better sense that the user is using a routine that works today but is expected to go away in the near future.  The current wording suggests that you did something wrong, but then the code still works which seems surprising/confusing.

Thanks for remembering to add something to test/deprecated.

**Comment by mppf** [comment]
Hi @bradcray - not a problem. Please see #5329 which tries to improve the wording. Let me know if you are happy with that wording or if you have other suggestions.

**Pull request merged**